test-d/record.test-d.ts

```diff
@@ -35,3 +35,18 @@ const data: unknown = null
 
   expectType<{ a: { b: { c: string } } }>(rt(data))
 }
+
+// record with json
+{
+  const rt = st.record({
+    a: st.string(),
+    b: st.json(
+      st.record({
+        c: st.string(),
+      }),
+    ),
+    d: st.string(),
+  })
+
+  expectType<{ a: string; b: { c: string }; d: string }>(rt(data))
+}
```

Comment:
Because the tests in `test-d/json.test-d.ts` check that `json` produces the correct types, this test is not strictly necessary, so I can remove if that is preferred. I also know it's impractical to have tests for all the possible interactions between runtypes. However, I added this one as it is common in my use case (e.g. `record` that contains a `json`, which then contains a `record`), but I can remove if that is preferred.